core/src/main/java/jenkins/model/GlobalConfiguration.java

```diff
@@ -78,11 +84,38 @@ public boolean configure(StaplerRequest req, JSONObject json) throws FormExcepti
     }
 
     /**
-     * Assume by default that GlobalConfiguration items require Jenkins.ADMINISTER permission
+     * Returns the permission type needed in order to use/access this
+     * By default,  require Jenkins.ADMINISTER permission
      * Ovveride to return something different if appropriate
-     * @return Permission
+     * @return Permission requiered to use/access this
      */
     public Permission getPermission() {
         return Jenkins.ADMINISTER;
     }
+
+    @Extension
+    public static class GlobalConfigHiddenByPermissionFilter extends DescriptorVisibilityFilter {
+
+        @SuppressWarnings("rawtypes")
+        @Override
+        public boolean filter(Object context, Descriptor descriptor) {
+            boolean result = false;
+            try {
+                if (descriptor instanceof GlobalConfiguration) {
+                    result = Functions.hasPermission(((GlobalConfiguration) descriptor).getPermission());
```

Comment:
you can just return the boolean value instead of assigning it to return it at the end. It's a simpler solution.